packages/launcher/src/index.tsx

```diff
@@ -187,25 +381,63 @@ export class Launcher extends VDomRenderer<LauncherModel> {
       }
     }
 
+    let tableCategories = Object.create(null);
+    for (let cat in categories) {
+      tableCategories[cat] = [...categories[cat]];
+    }
+
+    let orderedItems = this.model.sortedItemsByUsage();
+    let i = 0;
+    let cnt = 0;
+    let topUsed: ILauncher.IGroupedItemOptions[] = [];
+    while (cnt < 4 && i < orderedItems.length) {
```

Comment:
I didn't get the usage of the hard coded `4` here. Consider adding a comment to document.